libminifi/src/Site2SiteClientProtocol.cpp

```diff
@@ -682,6 +682,7 @@ bool Site2SiteClientProtocol::receive(std::string transactionID,
 
 bool Site2SiteClientProtocol::send(
     std::string transactionID, DataPacket *packet, std::shared_ptr<FlowFileRecord> flowFile,
+	uint8_t *payload, int length,
```

Comment:
As someone reading this that does not make sense. Why not make DataPacket hold the "packet"?